src/plugins/kibana_utils/public/history/get_query_params.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { parse, ParsedQuery } from 'query-string';
+import { Location } from 'history';
+
+export function getQueryParams(location: Location): ParsedQuery {
```

Comment:
If we make this generic, can we avoid the cast `as string` above?

```suggestion
export function getQueryParams<T>(location: Location): ParsedQuery<T> {
```

Reply:
Looked into it, 
`ParsedQuery<T>` is a weird loose type:

```
export interface ParsedQuery<T = string> {
	[key: string]: T | T[] | null;
}
```

that attempt to cover all the query cases and doesn't cover `undefined`. 
T in that case is just type of `value` in query object.

So I guess possible way to improve with typecasting for us is something like:

```
export function getQueryParams<T>(location: Location): T {
 return query as unknown as T
}
```

But I think this is worth then current downcasting on consumer level, because this involves `unknown`